cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/StateVariablesAdder.java

```diff
@@ -127,22 +325,106 @@ private static PropertyBag createPowerFlowProperties(CgmesModel cgmes, Terminal
+            // TopologicalNodes.
+            // Also, there can be > 1 TopologicalIsland, so we need to re-group PropertyBags
+            // by TopologicalIsland ID.
+            Map<String, PropertyBags> byTopologicalIslandId = new HashMap<>();
+            originalTopologicalIslands.forEach(pb -> {
+                String island = pb.getId(CgmesNames.TOPOLOGICAL_ISLAND);
+                if (byTopologicalIslandId.keySet().contains(island)) {
+                    byTopologicalIslandId.get(island).add(pb);
+                } else {
+                    PropertyBags pbs = new PropertyBags();
+                    pbs.add(pb);
+                    byTopologicalIslandId.put(island, pbs);
+                }
+            });
+            // now we can process all TPNodes from each island, and put them in one
+            // multivaluedProperty.
+            PropertyBags topologicalIslands = new PropertyBags();
+            byTopologicalIslandId.values().forEach(island -> {
+                PropertyBag topologicalIsland = new PropertyBag(SV_TOPOLOGICALISLAND_PROPERTIES);
+                topologicalIsland.setClassPropertyNames(Arrays.asList(CgmesNames.NAME));
+                topologicalIsland.setMultivaluedProperty(Arrays.asList("TopologicalNodes"));
+                topologicalIsland.put(CgmesNames.NAME, island.get(0).getId("name"));
+                topologicalIsland.put(CgmesNames.ANGLEREF_TOPOLOGICALNODE,
+                    island.get(0).getId(CgmesNames.ANGLEREF_TOPOLOGICALNODE));
+                topologicalIsland.put(CgmesNames.TOPOLOGICAL_NODES,
+                    String.join(",", island.pluckLocals(CgmesNames.TOPOLOGICAL_NODES)));
+                topologicalIslands.add(topologicalIsland);
+            });
+            cgmes.add(originalSVcontext, CgmesNames.TOPOLOGICAL_ISLAND, topologicalIslands);
+        }
+    }
+
+    // Added full model data with proper profile (StateVariables)
+    // FullModel is defined in ModelDescription:
+    // http://iec.ch/TC57/61970-552/ModelDescription/1#
+    private void addModelDescriptionToCgmes() {
+        PropertyBags fullModelSV = new PropertyBags();
+        // for properties such as "md:Model.DependentOn" which might have arbitrary
+        // number of values, SPARQL will return multiple result sets.
+        // All are equal, except the multiValued property value.
+        if (!originalFullModel.isEmpty()) {
+            PropertyBag newModelDescription = new PropertyBag(SV_FULLMODEL_PROPERTIES);
+            newModelDescription
+                .setClassPropertyNames(
+                    Arrays.asList(CgmesNames.SCENARIO_TIME, CgmesNames.CREATED, CgmesNames.DESCRIPTION,
+                        CgmesNames.VERSION, CgmesNames.DEPENDENT_ON, CgmesNames.PROFILE,
+                        CgmesNames.MODELING_AUTHORITY_SET));
+            newModelDescription.setMultivaluedProperty(Arrays.asList(CgmesNames.DEPENDENT_ON));
```

Comment:
You can replace `Arrays.asList(CgmesNames.DEPENDENT_ON)` by `Collections.singletonList(...)`

Reply:
Done